pymeasure/display/widgets/dock_widget.py

```diff
@@ -48,14 +51,16 @@ class DockWidget(TabWidget, QtWidgets.QWidget):
         than x_axis_labels the last item in the list to match x_axis_labels length.
     :param linewidth: line width for plots in
         :class:`~pymeasure.display.widgets.plot_widget.PlotWidget`
+    :param layout_path: Directory path to save dock layout state. Default is './'
     :param parent: Passed on to QtWidgets.QWidget. Default is None
     """
 
     def __init__(self, name, procedure_class, x_axis_labels=None, y_axis_labels=None, linewidth=1,
-                 parent=None):
+                 layout_path='./', parent=None):
         super().__init__(name, parent)
 
         self.procedure_class = procedure_class
+        self.dock_layout_filename = layout_path + procedure_class.__name__ + '_dock_layout.json'
```

Comment:
Maybe (see my general comment) an option would be to have  `layout_filename=None` optional kwargs, and only use the default `procedure_class.__name__ + '_dock_layout.json'` if layout_filename is not provided:
```suggestion
        self.dock_layout_filename = path.join(layout_path, layout_filename or
                                              (procedure_class.__name__ + '_dock_layout.json'))
```

Though I'm not entirely sure if this would be all too useful.

Reply:
Implemented in last commit